core/vm/interpreter.go

```diff
@@ -204,25 +213,32 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 		res     []byte // result of the opcode execution function
 		debug   = in.evm.Config.Tracer != nil
 	)
+	in.pc = 0
+	in.ctx = ScopeContext{
```

Comment:
Would it make sense to to move `Memory` `Stack` up to the EVMInterpreter/runContext

Reply:
They are kinda already in `EVMInterpreter/runContext/ScopeContext` as pointers.

There is not much benefit in puttting them in `runContext` as values like `Contract`, since they are managed by a pool. If we did, when we are releasing them back to pool, we would have to keep allocating heap Memory/Stack instances and copy the value in runContext to that newly allocated heap instance.